x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/settings/UpdateSecuritySettingsAction.java

```diff
@@ -0,0 +1,141 @@
+            this.mainIndexSettings = Objects.requireNonNullElse(mainIndexSettings, Collections.emptyMap());
+            this.tokensIndexSettings = Objects.requireNonNullElse(tokensIndexSettings, Collections.emptyMap());
+            this.profilesIndexSettings = Objects.requireNonNullElse(profilesIndexSettings, Collections.emptyMap());
+        }
+
+        public Request(StreamInput in) throws IOException {
+            this.mainIndexSettings = in.readMap();
+            this.tokensIndexSettings = in.readMap();
+            this.profilesIndexSettings = in.readMap();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeGenericMap(this.mainIndexSettings);
+            out.writeGenericMap(this.tokensIndexSettings);
+            out.writeGenericMap(this.profilesIndexSettings);
+        }
+
+        public static Request parse(XContentParser parser) {
+            return PARSER.apply(parser, null);
+        }
+
+        public Map<String, Object> mainIndexSettings() {
+            return this.mainIndexSettings;
+        }
+
+        public Map<String, Object> tokensIndexSettings() {
+            return this.tokensIndexSettings;
+        }
+
+        public Map<String, Object> profilesIndexSettings() {
+            return this.profilesIndexSettings;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = validateIndexSettings(mainIndexSettings, MAIN_INDEX_NAME, null);
+            validationException = validateIndexSettings(tokensIndexSettings, TOKENS_INDEX_NAME, validationException);
+            validationException = validateIndexSettings(profilesIndexSettings, PROFILES_INDEX_NAME, validationException);
+            if (validationException != null) return validationException;
+            return null;
+        }
+
+        private static ActionRequestValidationException validateIndexSettings(
+            Map<String, Object> indexSettings,
+            String indexName,
+            ActionRequestValidationException existingExceptions
+        ) {
+            if (indexSettings.containsKey("index") == false) {
+                // return ValidateActions.addValidationError("only [index] settings may be set")
+            }
+            Set<String> forbiddenSettings = Sets.difference(indexSettings.keySet(), ALLOWED_SETTING_KEYS);
+            if (forbiddenSettings.size() > 0) {
+                return ValidateActions.addValidationError(
+                    "illegal settings for index ["
+                        + indexName
+                        + "]: "
+                        + forbiddenSettings
+                        + ", these settings may not be configured. Only the following settings may be configured for that index: "
+                        + ALLOWED_SETTING_KEYS,
+                    existingExceptions
+                );
+            }
+            return null;
```

Comment:
Shouldn't we return `existingExceptions` here? Otherwise, `existingExceptions` gets swallowed.

Reply:
Uh yep good catch, thanks!